src/main/java/org/pentaho/di/trans/dataservice/clients/AnnotationsQueryService.java

```diff
@@ -97,7 +97,7 @@ private void writeAnnotations( final OutputStream outputStream, final ModelAnnot
       rowMeta.addValueMeta( new ValueMetaString( "annotations" ) );
       rowMeta.writeMeta( dos );
 
-      Object[] row = new Object[] { writer.getXML() };
+      Object[] row = new Object[] { writer.getXML().trim() };
```

Comment:
ModelAnnotationGroupXmlWriter returns document with empty <annotations><annotations/> tag in case no annotations passed.